jekyll/_posts/2018-06-09-discover-godot-nim.md

```diff
@@ -0,0 +1,93 @@
+_But!_ I did finish 2 small samples following the ["Your first game"](http://docs.godotengine.org/en/3.0/getting_started/step_by_step/your_first_game.html) tutorial and a [Conways Game of Life](http://www.brainjargames.com/game-of-life/) post using Nim instead of GDScript. Using types made things a lot clearer in code and I learned the awesome feeling of tackling a problem that might seem impossible to figure out at first, but to which the answer is so simple once solved.
+For those who are also keen to try Nim with Godot, I'd say take a look at the stub project and the api the bindings generate. My experience following the GDScript focused tutorials with Nim was that it all translates pretty well. Nim compiles so fast that I didn't find myself growing tired of the compile script -> press f5 in godot cycle.
+
+```nim
+import godot
+import animation_player, control
+
+type
+  CellKind* {.pure.} = enum # the asterisk means it's exported
+    Living
+    Dead
+
+gdobj Cell of Control:
+  var kind: CellKind
+  var x*, y*: int
+  var anim: AnimationPlayer
+
+  method ready*() =
+    anim = getNode("AnimationPlayer").as(AnimationPlayer)
+
+  proc initTile*(x,y: int) =
+    self.x = x
+    self.y = y
+
+  proc setKind*(kind: CellKind) =
+    self.kind = kind
+    case kind:
+    of CellKind.Living:
+      anim.play("Living")
+    of CellKind.Dead:
+      anim.play("Dead")
+
+  proc getKind*(): CellKind =
+    kind
+```
+
+##### A lot of thanks to @endragor, one of the maintainers of the plugin for answering my often stupid questions
+
+`gdobj` is the macro goodness that makes extending a node so easy. Declaring fields, properties and methods all happen inside it. Once compiled slap it on as a `NativeScript` and add your `nimlib.gdnlib` . I wrote the game of life sample first and then worked on the Dodge the Creeps project, the second time my workflow was _a lot_ faster. Hurray learning and doing!
+And everything you can do in GDScript you can do with Nim. Nim is also really fast.
+One thing I really did struggle with was registering a signal and emitting signals. I found the concept confusing reading about it but in GDScript it's a simple `signal foo`. For those who care in Nim you register a signal and emit a signal like this:
+
+```nim
+gdobj Foo of Node2D:
+   method init():
+       addUserSignal("Nim Rocks!") #register a signal
+       emitSignal("Godot Rocks!") #emit a signal
+```
+
+### Closing sentiments
+
+If you want to look at my _ahem_ work of art you can check it out **[here](http://docs.godotengine.org/en/3.0/getting_started/step_by_step/your_first_game.html#finishing-up).**
+Godot and Nim are both wonderful open source and community driven projects. I thoroughly enjoyed using them and using them together. As a beginner I didn't take the smartest route but I was more productive(and learned a heck more) in 2 weeks than I was following another Unity/Unreal course about how to make a ball bounce. So don't be afraid to jump in the pool!
```

Comment:
~~community driven~~ → community-driven
~~more productive(and learned a heck more)~~ → more productive (and learned a heck more)